File: app/src/main/java/org/fossasia/susi/ai/rest/clients/DeviceClient.java
package org.fossasia.susi.ai.rest.clients;

import org.fossasia.susi.ai.BuildConfig;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class DeviceClient {
    private static Retrofit retrofit = null;

    /**
     * Gets client.
     *
     * @return the client
     */
    public static Retrofit getClient() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BuildConfig.SPEAKER_BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/org/fossasia/susi/ai/rest/clients/DeviceClient.kt
package org.fossasia.susi.ai.rest.clients

import org.fossasia.susi.ai.BuildConfig
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object DeviceClient {
    val retrofit: Retrofit by lazy {
        Retrofit.Builder()
                .baseUrl(BuildConfig.SPEAKER_BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        }
}
